homeassistant/components/wittiot/sensor.py

```diff
@@ -0,0 +1,1207 @@
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.HUMIDITY,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LEAFCH1_BATT,
+        sensor_type=LEAF_BATT,
+        name="CH1 Leaf Battery",
+        icon="mdi:battery",
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LEAFCH2_BATT,
+        sensor_type=LEAF_BATT,
+        name="CH2 Leaf Battery",
+        icon="mdi:battery",
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LEAFCH3_BATT,
+        sensor_type=LEAF_BATT,
+        name="CH3 Leaf Battery",
+        icon="mdi:battery",
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LEAFCH4_BATT,
+        sensor_type=LEAF_BATT,
+        name="CH4 Leaf Battery",
+        icon="mdi:battery",
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LEAFCH5_BATT,
+        sensor_type=LEAF_BATT,
+        name="CH5 Leaf Battery",
+        icon="mdi:battery",
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LEAFCH6_BATT,
+        sensor_type=LEAF_BATT,
+        name="CH6 Leaf Battery",
+        icon="mdi:battery",
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LEAFCH7_BATT,
+        sensor_type=LEAF_BATT,
+        name="CH7 Leaf Battery",
+        icon="mdi:battery",
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LEAFCH8_BATT,
+        sensor_type=LEAF_BATT,
+        name="CH8 Leaf Battery",
+        icon="mdi:battery",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Wittiot sensor entities based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    sensors = []
+    for desc in SENSOR_DESCRIPTIONS:
+        if coordinator.data.get(desc.key) not in ("", "--", "--.-", "None"):
+            sensors.append(
+                LocalWittiotSensor(
+                    coordinator,
+                    entry.data[CONF_IP],
+                    entry.data[DEVICE_NAME],
+                    desc,
+                )
+            )
+    async_add_entities(sensors, True)
+
+
+class LocalWittiotSensor(CoordinatorEntity[WittiotDataUpdateCoordinator], SensorEntity):
+    """Define an Local sensor."""
+
+    _attr_has_entity_name = True
+    entity_description: WittiotSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: WittiotDataUpdateCoordinator,
+        ipadrr: str,
+        devname: str,
+        description: WittiotSensorEntityDescription,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"{devname}_{description.sensor_type}")},
+            manufacturer="WittIOT",
+            name=f"{devname}_{description.sensor_type}",
+            model=self.coordinator.data["ver"],
+            configuration_url=f"http://{ipadrr}",
+        )
```

Comment:
Last time you said you were looking into making different devices for each meter, what's up with that

Reply:
I added a 'sensor_type' parameter to distinguish the different types of devices, and now the different types of data in the panel are displayed in their respective forms.

Reply:
Can we split them on a physical device instead of type?

Reply:
I hope to be able to split by type, because I found in the local test that after adding new devices, the data of the sensor is classified and displayed on the homepage, and the default display effect looks very good.

Reply:
No. You told me that every channel is their own measuring point and its own physical device. Thus one channel is one device in HA. This also allows the user to rename the device to maybe where they placed the sensor to give it more context.

Reply:
Can I change it this way?
The identifiers are bound to the key of the sensor, which is different for each sensor, so that each entity will be a device.

Reply:
No. This is how our devices work as you can read in our dev docs. This will make this PR way smaller, consistent with the rest of the integrations and easier to manage

Reply:
Can you give me some hints on how to implement this in the code?

Reply:
If i understand correctly you have the main device, and sub devices. Give them all a different identifier and for the sub devices make sure you add the `via_device` and set it to the main device. I think checking the Dev docs on this is the best way. And if you have trouble, let me know and I can think of an example integration for your problem, can't think of any now